src/frontend/state/libraryState.ts

```diff
@@ -0,0 +1,375 @@
+        )
+          return true
+      } else {
+        if (game.is_windows_native) return true
+      }
+    }
+
+    if (platformsToInclude.includes('mac')) {
+      if (game.is_installed) {
+        if (
+          game.install.platform &&
+          macArray.includes(getPlatformName(game.install.platform))
+        )
+          return true
+      } else {
+        if (game.is_mac_native) return true
+      }
+    }
+
+    if (platformsToInclude.includes('linux')) {
+      if (game.is_installed) {
+        if (
+          game.install.platform &&
+          getPlatformName(game.install.platform).toLowerCase() === 'linux'
+        )
+          return true
+      } else {
+        if (game.is_linux_native) return true
+      }
+    }
+
+    if (game.browserUrl && platformsToInclude.includes('browser')) return true
+
+    return false
+  }
+
+  filterByPlatform(library: GameInfo[], platformsToInclude: Platform[]) {
+    if (!library) {
+      return []
+    }
+
+    return library.filter((game) =>
+      this.gameSupportsAtLeastOneFilteredPlatform(game, platformsToInclude)
+    )
+  }
+
+  get favourites() {
+    const tempArray: GameInfo[] = []
+    if (this.showFavourites || this.showFavouritesLibrary) {
+      const favouriteAppNames = this.favouriteGames?.list.map(
+        (favourite: FavouriteGame) => favourite.appName
+      )
+      if (favouriteAppNames === undefined) return tempArray
+      this.epicLibrary.forEach((game) => {
+        if (favouriteAppNames.includes(game.app_name)) tempArray.push(game)
+      })
+      this.gogLibrary.forEach((game) => {
+        if (favouriteAppNames.includes(game.app_name)) tempArray.push(game)
+      })
+      this.sideloadedLibrary.forEach((game) => {
+        if (favouriteAppNames.includes(game.app_name)) tempArray.push(game)
+      })
+      this.hyperPlayLibrary.forEach((game) => {
+        if (favouriteAppNames.includes(game.app_name)) tempArray.push(game)
+      })
+    }
+    return tempArray
+  }
+
+  // automatically inferred as computed by makeAutoObservable
+  get library() {
+    let library: Array<GameInfo> = []
+    if (this.showFavouritesLibrary) {
+      library = [...this.favourites].filter((g) =>
+        this.category === 'all' ? g : g.runner === this.category
+      )
+    } else {
+      const isEpic = epicCategories.includes(this.category)
+      const isGog = gogCategories.includes(this.category)
+      const epicLibrary = isEpic ? this.epicLibrary : []
+      const gogLibrary = isGog ? this.gogLibrary : []
+      const sideloadedApps = sideloadedCategories.includes(this.category)
+        ? this.sideloadedLibrary
+        : []
+      const HPLibrary = hyperPlayCategories.includes(this.category)
+        ? this.hyperPlayLibrary
+        : []
+
+      library = [...HPLibrary, ...sideloadedApps, ...epicLibrary, ...gogLibrary]
+
+      if (!this.showNonAvailable) {
+        // TODO: refactor to linear algo with non available game map
+        library = library.filter(
```

Comment:
copilot for the win:
```typescript
const nonAvailableAppNames = Object.fromEntries(this.nonAvailbleGames.map((game) => [game.app_name, true]))
library = library.filter((game) => !nonAvailableAppNames[game.app_name])
```

Reply:
nice. this is pretty clean. and linear :partying_face: just updated it

and we can confidently make changes like this now because we have tests :man_dancing: